stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -282,21 +282,6 @@ internal constructor(
                         setUser(user)
                     }
                 }
-                is DisconnectedEvent -> {
-                    when (event.disconnectCause) {
-                        DisconnectCause.ConnectionReleased,
-                        DisconnectCause.NetworkNotAvailable,
-                        is DisconnectCause.Error,
-                        -> if (ToggleService.isSocketExperimental().not()) socketStateService.onDisconnected()
-                        is DisconnectCause.UnrecoverableError -> {
-                            userStateService.onSocketUnrecoverableError()
-                            if (ToggleService.isSocketExperimental().not()) {
-                                socketStateService.onSocketUnrecoverableError()
-                            }
-                        }
-                    }
-                    clientState.toMutableState()?.setConnectionState(ConnectionState.OFFLINE)
-                }
                 is NewMessageEvent -> {
```

Comment:
I wonder why this was extracted to a separate observer? 🤔

Reply:
Tbh, I cannot understand how it may guarantee that if Socket notifies about coming events using `Handler`:
```
eventUiHandler.post { call(listener) }
```

Reply:
When I wrote the reply there were notifications with and without `Handler` and this one was not using then. But you're right, we can't use `Handler` if we would like to make sure that the notification has already happened.

Reply:
I did some testing and using `subscribeSuspend` makes the socket listener to run after the Result.success is complete. But that doesn't happen when using `subscribe`, using a coroutine call makes it wait to the listener to complete.